### ISSUE TITLE
AG-Grid Accessibility Issue - Elements should have focusable content

### ISSUE THREAD

**Issue by Arpita-Blog** [ag-grid/ag-grid#7287]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Customers of AG Grid**

**Providing a Reproducible Scenario**
I have attached a codesandbox link below to reproduce the issue 
https://codesandbox.io/p/sandbox/stoic-darkness-3d89zr?file=%2Fsrc%2FApp.tsx%3A65%2C27

Steps to reproduce the issue 
1) Open the codesandbox link, run the application
2) Enable fast pass application to run the accessibility test

**Current behavior**
Ag-grid is throwing an accessibility issue stating that an element should be focusable.

![image](https://github.com/ag-grid/ag-grid/assets/90197139/1c25a984-8666-41ef-8c41-767520921c45)
![image](https://github.com/ag-grid/ag-grid/assets/90197139/ea2a96be-5614-4063-9c0b-267b47062b62)

**Expected behavior**
When using ag-Grid, the elements in the grid should be properly focusable according to accessibility standards. There should be no issues or errors related to element focus, and users should be able to navigate through the grid using keyboard controls without encountering any accessibility violations

* **AG Grid version:** X.X.X
"ag-charts-react": "^7.2.0",
"ag-grid-community": "^30.0.0",
"ag-grid-enterprise": "^30.0.0",
"ag-grid-react": "^30.0.0",

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by AG-Zoheil** [comment]
Hi @Arpita-Blog,

Thank you for submitting this.
 
The accessibility audit tools base their results on static analysis of the DOM generated by AG Grid. This dictates that an element with scrollable content is accessible by keyboard.
 
AG Grid uses its own custom container to provide row and column virtualization to allow showing an unlimited number of rows in the grid while recycling a limited number of actual DOM elements to show the visible rows. This custom container doesn't use the default focus and scrolling logic that non-virtualised elements use and that accessibility audit tools expect. In order to support virtualised lists, AG Grid manages focus and keyboard navigation using its own custom focus, scrolling and keyboard navigation code.
 
Accessibility audit tools are not aware of the custom code to handle focus, scrolling and keyboard navigation across AG Grid and flag this as an issue. However, in fact there are no issues in scrolling/focus/keyboard navigation in AG Grid.

I hope this helps.

Kind regards,
Zoheil
